controller/state.go

```diff
@@ -79,12 +79,11 @@ func (ks *KsonnetAppStateManager) groupLiveObjects(liveObjs []*unstructured.Unst
 	return liveByFullName
 }
 
-// CompareAppState compares application spec and real app state using KSonnet
-func (ks *KsonnetAppStateManager) CompareAppState(app *v1alpha1.Application) (*v1alpha1.ComparisonResult, *repository.ManifestResponse, error) {
-	repo := ks.getRepo(app.Spec.Source.RepoURL)
-	conn, repoClient, err := ks.repoClientset.NewRepositoryClient()
+// GetManifest retrieves a manifest from the given repo and app version.
+func GetManifest(repoClientset reposerver.Clientset, repo *v1alpha1.Repository, app *v1alpha1.Application) (*repository.ManifestResponse, error) {
```

Comment:
Its not necessary to break apart this function. Application server already has repoClientset and can query the repo server directly.

Reply:
This was my most recent change.  Reverting `HEAD`.